<launch>
  <!-- Mainly copied from -->
  <!-- https://github.com/jsk-ros-pkg/jsk_recognition/blob/master/jsk_pcl_ros/sample/sample_tilt_laser_listener.launch -->

  <arg name="laser_cloud" default="/tilt_laser_listener/large_cloud" />

  <arg name="manager" default="tilt_laser_manager" />
  <node name="$(arg manager)" pkg="nodelet" type="nodelet" args="manager"/>

  <node name="laser_scan_assembler"
        pkg="laser_assembler" type="laser_scan_assembler">
    <remap from="scan" to="/tilt_scan"/>
    <remap from="assemble_scans2" to="~assemble_scans2"/>
    <rosparam>
      fixed_frame: base_link
      max_scans: 400
    </rosparam>
  </node>

  <node name="tilt_laser_listener"
        pkg="nodelet" type="nodelet"
        args="load jsk_pcl/TiltLaserListener $(arg manager)">
    <remap from="~input" to="/joint_states"/>
    <remap from="~assemble_scans2" to="laser_scan_assembler/assemble_scans2"/>
    <rosparam>
      use_laser_assembler: true
      not_use_laser_assembler_service: false
      joint_name: laser_tilt_mount_joint
      twist_frame_id: laser_tilt_link
      laser_type: tilt
    </rosparam>
  </node>

  <node name="extract_large_pointcloud" pkg="kitchen_registration"
        type="extract_large_pointcloud.py">
    <remap from="~output" to="$(arg laser_cloud)" />
    <rosparam>
      duration: 30 <!-- how many seconds to collect laser pointcloud -->
    </rosparam>
  </node>
</launch>
